scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/NameDetectAnalyzer.kt

```diff
@@ -0,0 +1,268 @@
+import android.media.MediaScannerConnection
+import android.media.MediaScannerConnection.OnScanCompletedListener
+import android.provider.MediaStore
+import android.util.Log
+import com.getbouncer.scan.framework.Analyzer
+import com.getbouncer.scan.framework.AnalyzerFactory
+import com.getbouncer.scan.framework.ml.ssd.hardNonMaximumSuppression
+import com.getbouncer.scan.payment.ml.AlphabetDetect
+import com.getbouncer.scan.payment.ml.SSDObjectDetect
+import com.getbouncer.scan.payment.ml.SSDOcr
+import com.getbouncer.scan.payment.ml.ssd.DetectionBox
+import com.getbouncer.scan.payment.ml.toObjectDetectionCroppedImage
+import kotlinx.coroutines.CoroutineScope
+import java.util.*
+
+
+private const val NUM_PREDICTION_STRIDES = 10
+private const val NMS_THRESHOLD = 0.75F
+private const val CHAR_CONFIDENCE_THRESHOLD = 0.7
+
+
+class NameDetectAnalyzer private constructor(
+    private val ssdObjectDetect: SSDObjectDetect?,
+    private val alphabetDetect: AlphabetDetect?
+) : Analyzer<SSDOcr.SSDOcrInput, PaymentCardOcrState, NameDetectAnalyzer.Output> {
+
+    data class Output(
+        val name: String?,
+        val boxes: List<DetectionBox>?
+    )
+
+    override val name: String = "card_side_analyzer"
+
+    override suspend fun analyze(data: SSDOcr.SSDOcrInput, state: PaymentCardOcrState): Output {
+        return if (state.runNameExtraction && ssdObjectDetect != null && alphabetDetect != null) {
+            val objDetectInput = SSDObjectDetect.Input(
+                fullImage = data.fullImage,
+                previewSize = data.previewSize,
+                cardFinder = data.cardFinder,
+                iin = null
+            )
+
+            val output = ssdObjectDetect.analyze(objDetectInput, Unit)
+            var nameRect: RectF? = null
+
+            // check to see if we detected a legal name. If so, run our name extraction model, serially for now
+            for (box in output.detectionBoxes) {
+                if (box.label == SSDObjectDetect.Labels.NAME.ordinal) {
+                    nameRect = box.rect
+                    break
+                }
+            }
+
+            val name = if (nameRect != null) {
+                processPredictions(nameRect, objDetectInput.toObjectDetectionCroppedImage())
+            } else {
+                null
+            }
+
+            Output(name, output.detectionBoxes)
+        } else {
+            Output("", null)
+        }
+    }
+
+    internal class CharPredictionWithBox(
+        val characterPrediction: AlphabetDetect.Prediction,
+        val box: RectF
+    ) {
+        private val boxArray: FloatArray = FloatArray(4)
+
+        fun getPercentBoxArray(width: Int, height: Int): FloatArray {
+            val pctBoxArray = FloatArray(4)
+            pctBoxArray[0] = boxArray[0] / width
+            pctBoxArray[1] = boxArray[1] / height
+            pctBoxArray[2] = boxArray[2] / width
+            pctBoxArray[3] = boxArray[3] / height
+            return pctBoxArray
+        }
```

Comment:
I've been trying to avoid the use of arrays of values where possible for performance, and instead use objects with named fields. This makes understanding the logic behind these processes easier.

```suggestion
        fun getPercentBoxArray(width: Int, height: Int) = RectF(
            box.left / width,
            box.top / height,
            box.right / width,
            box.bottom / height
        )
```